src/util/uri.rs

```diff
@@ -6,13 +6,13 @@ use crate::error::Error;
 
 use std::marker::PhantomData;
 
-use reqwest::blocking::Client;
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use ureq::{Agent, Response};
 
 /// A URI that will fetch something of a defined type `T`.
 #[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
 #[serde(transparent)]
-pub struct URI<T>(String, PhantomData<T>);
+pub struct URI<T>(String, PhantomData<fn() -> T>);
```

Comment:
Why did you make this change? Looks fancy. I've seen this before but I don't know if I understand it

Reply:
It has to do with what you're indicating and covariance. Adding a `PhantomData<T>` means that the type conceptually owns a `T`. Using `PhantomData<fn() -> T>` means the type produces owned `T`s. The difference between these is that the drop checker analyzes the lifetime of `T` in the former case. [Here's a list of `PhantomData` patterns.](https://doc.rust-lang.org/nomicon/phantom-data.html#table-of-phantomdata-patterns)

There's probably no functional difference with your usage here, but it's just something I noticed.